src/misc/ansi_code_markup.cpp

```diff
@@ -0,0 +1,200 @@
+	        {"black", 30},  {"red", 31},   {"green", 32},
+	        {"yellow", 33}, {"blue", 34},  {"magenta", 35},
+	        {"cyan", 36},   {"white", 37}, {"default", 39},
+	};
+};
+
+class TagParser {
+public:
+	TagParser() : color() {}
+	/*!
+	 * \brief Returns a pointer to an ANSI code
+	 *
+	 * The returned string is owned by this object. It remains valid until
+	 * the next call to this function.
+	 *
+	 * \param tag tag name (eg: b, color)
+	 * \param close whether this is a closing tag
+	 * \param color_val the color value if a color tag
+	 * \return const char*
+	 */
+	const char *get_ansi_code(std::string &tag, bool close, std::string &color_val)
+	{
+		if (tag == "color" || tag == "bgcolor") {
+			// We don't support closing color tags
+			if (close) {
+				return nullptr;
+			}
+			return color.get_ansi_code(color_val, tag == "bgcolor");
+		}
+		reset_str(ansi_code);
+		if (!map_contains<std::string, int>(tags, tag)) {
+			return nullptr;
+		}
+		int ansi_num = tags.at(tag);
+		if (close) {
+			// "closing" tags have ascii codes +20
+			ansi_num += 20 + (tag == "b" ? 1 : 0); // [/b] is the
+			                                       // same as [/dim]
+		}
+		safe_sprintf(ansi_code, "\033[%dm", ansi_num);
+		return ansi_code;
+	}
+
+private:
+	ColorParser color;
+	char ansi_code[10] = {0};
+	std::unordered_map<std::string, int> tags = {
+	        {"reset", 0}, {"b", 1},       {"dim", 2},
+	        {"i", 3},     {"u", 4},       {"s", 9},
+	        {"blink", 5}, {"inverse", 7}, {"hidden", 8},
+	};
+};
+
+static TagParser tag_parser;
+
+static std::regex markup(R"((\\)?)" // Escape tag?
+						 "(\\["      // Opening bracket, open main group
+                         "[ \\t]*?" // Optional spacing after opening bracket
+                         "(\\/)?"   // Check for losing tag
+                         "("        // Start group of tags
+                         "((?:bg)?color)" // Select color or bgcolor. bg not
+                                          // captured in separate group
+                         "(?:[ \\t]*?=[ \\t]*?([a-z\\-]+))?" // Color value. '='
+                                                             // not captured in
+                                                             // separate group.
+                                                             // Spacing around
+                                                             // '=' is allowed
+                         "|i|b|u|s|blink|dim|hidden|inverse|reset" // All other
+                                                                   // tags to match
+                         ")"        // End group of tags
+                         "[ \\t]*?" // Optional spacing before closing bracket
+                         "\\])" // Closing bracket. Note: Clang and MSVC requires
+                               // this to be escaped. Closing main group
+                         ,
+                         std::regex::icase);
```

Comment:
excellent documentation per component!
If you can add a comment at the top just showing an example "sunny day string" that it pulls apart, that will help readers connect each regex component to its respective string part.